core/src/main/java/com/nextdoor/bender/handler/HandlerMetadata.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright $year Nextdoor.com, Inc
+ */
+
+package com.nextdoor.bender.handler;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import sun.reflect.generics.reflectiveObjects.NotImplementedException;
+
+/**
+ * The {@link HandlerMetadata} object is a simple Key/Value map that is used by {@Link
+ * BaseHandler} to store information about how and when the function was invoked. This data is
+ * either thrown away at the end of the execution, or is used later by our
+ * {@link com.nextdoor.bender.wrapper.Wrapper} and
+ * {@link com.nextdoor.bender.operation.Operation} classes to futher modify the payload before
+ * its shipped.
+ */
+public class HandlerMetadata {
+    private Map<String, Object> payload;
+    private Boolean immutable = false;
+
+    public HandlerMetadata() {
+        this.payload = new HashMap<String, Object>();
+    }
+
+    public Map<String, Object> getPayload() {
+      return payload;
+    }
+
+    public Object getField(String fieldName) throws NoSuchElementException {
+      if (this.payload == null) {
+          throw new NoSuchElementException(fieldName + " is not in payload because payload is null");
+      }
+
+      Object o = this.payload.getOrDefault(fieldName, null);
```

Comment:
Since you throw a `NoSuchElementException` above would it make sense to differentiate not present from `null` when asking for a field that may not exist or may be null?

Reply:
Yeah the reality here is that this code path basically cannot be hit anyways.

Reply:
I've removed that bit of code. The `getField` method here is mostly used for test/validation anyways, its not used in any main code paths. Returning Null feels fine to me if the value doesn't exist. I can't see a reason why you would intentionally set a piece of metadata to `null`?